main/accounts/Account/index.ts

```diff
@@ -257,7 +257,7 @@ class FrameAccount {
 
       this.addRequiredApproval(
         req,
-        ApprovalType.TokenSpendApproval,
+        amount === '0x00' ? ApprovalType.TokenSpendRevocation : ApprovalType.TokenSpendApproval,
```

Comment:
I think we should make this check more flexible, maybe something like `parseInt(amount, 16) === 0` just in case we ever change how we represent the amount